File: mobileBE/src/main/java/com/uth/mobileBE/dto/request/UpdateLoanDetailRequest.java
package com.uth.mobileBE.dto.request;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class UpdateLoanDetailRequest {
    @NotNull(message = "Mã bản sao sách không được để trống")
    private Long copyId;

    @NotNull(message = "Trạng thái không được để trống")
    private String status; // BORROWING, RETURNED, LOST

    private LocalDateTime dueDate;

    private String condition;
}

File: app/src/main/java/com/example/quanlythuvien/data/model/request/UpdateLoanDetailRequest.kt
package com.example.quanlythuvien.data.model.request

import com.google.gson.annotations.SerializedName

data class UpdateLoanDetailRequest(
    @SerializedName("copyId") val copyId: Long,
    @SerializedName("status") val status: String,
    @SerializedName("dueDate") val dueDate: String?,
    // THÊM DÒNG NÀY ĐỂ GỬI TÌNH TRẠNG SÁCH (NEW, GOOD, FAIR, POOR)
    @SerializedName("condition") val condition: String? = null
)